src/main/java/com/checkmarx/flow/service/MergeResultEvaluatorImpl.java

```diff
@@ -155,8 +155,9 @@ private static void logThresholdCheck(
             if (rawSummary instanceof Map) {
                 result = (Map<?, ?>) rawSummary;
             } else {
+                String summaryClass = rawSummary != null ? rawSummary.getClass().getName() : null;
                 log.warn("Wrong summary type in scan results. Expected {}, got {}.",
-                        Map.class.getName(), rawSummary.getClass().getName());
+                        Map.class.getName(), summaryClass);
```

Comment:
Why do you care checking the value of summaryClass for null if in the end you treat the results same way on the log.warn?

Reply:
If rawSummary is null, rawSummary.getClass() will throw an exception (I've already stumbled upon this :).